.github/workflows/build-terraform-cli.yml

```diff
@@ -51,24 +51,29 @@ jobs:
           GO_LDFLAGS: ${{ inputs.ld-flags }}
           ACTIONSOS: ${{ inputs.runson }}
           CGO_ENABLED: ${{ inputs.cgo-enabled }}
-        uses: hashicorp/actions-go-build@v0.1.7
+        uses: hashicorp/actions-go-build@e20c6be7bf010e40e930dab20e6da63176725ec1 # v0.1.9
         with:
           product_name: ${{ inputs.package-name }}
           product_version: ${{ inputs.product-version }}
           go_version: ${{ inputs.go-version }}
           os: ${{ inputs.goos }}
           arch: ${{ inputs.goarch }}
           reproducible: nope
+          zip_name: ${{ env.ARTIFACT_BASENAME }}
           instructions: |-
-            mkdir dist out
-            set -x
-            go build -ldflags "${{ inputs.ld-flags }}" -o dist/ .
-            zip -r -j out/${{ env.ARTIFACT_BASENAME }} dist/
+            go build -ldflags "${{ inputs.ld-flags }}" -o "$BIN_PATH" -trimpath -buildvcs=false
+            cp LICENSE "$TARGET_DIR/LICENSE.txt"
```

Comment:
This is the part that I'm least confident about; as I was going through https://github.com/hashicorp/actions-go-build/blob/main/README.md I realized our implementation is wildly different from all the examples there. Notably, we currently run the `zip` command manually while the action seems to be intended to abstract that away. It seems like a good idea to me to be more consistent with how the action is intended to be used, but I worry a little that there might be unintended side effects.

Specific questions:
- Is it okay to remove the `set -x` in our `instructions`? Can I safely assume that the action is taking care of that?
- We don't currently use the `-trimpath -buildvcs=false` flags but all of the examples use them, should we be using them as well?
- Where does the zip file actually get created as part of this? (See also question on line     69 below.)

Reply:
> Is it okay to remove the set -x in our instructions? Can I safely assume that the action is taking care of that?
Yes this should be fine as` set -x` enables a mode of the shell where all executed commands are printed to the terminal for debugging purposes. If you specifically want to leave this in then that's also not a problem!

> We don't currently use the -trimpath -buildvcs=false flags but all of the examples use them, should we be using them as well?
You don't have to, this is used to help ensure that builds are reproducible (see https://github.com/hashicorp/actions-go-build?tab=readme-ov-file#build-path and https://github.com/hashicorp/actions-go-build?tab=readme-ov-file#vcs-information

> Where does the zip file actually get created as part of this? (See also question on line 69 below.)
the Zip file will automatically get [uploaded to the workflow run](https://github.com/hashicorp/actions-go-build/blob/80a04530485b66d1d36010dbd9c082886d7eb7b4/action.yml#L121) so you shouldn't need to do this in a separate step